docs/advanced/pool.md

```diff
@@ -0,0 +1,88 @@
+# Custom Pool
+
+::: warning
+This is advanced API. If you are just running tests, you probably don't need this. It is primarily used by library authors.
+:::
+
+Vitest runs tests in pools. By default, there are several pools:
+
+- `threads` to run tests using `node:worker_threads` (isolation is provided with a new worker context)
+- `forks` to run tests using `node:child_process` (isolation is provided with a new `child_process.fork` process)
+- `vmThreafs` to run tests using `node:worker_threads` (but isolation is provided with `vm` module instead of a new worker context)
```

Comment:
```suggestion
- `vmThreads` to run tests using `node:worker_threads` (but isolation is provided with `vm` module instead of a new worker context)
```